crates/ruma-api/src/error.rs

```diff
@@ -189,6 +189,7 @@ impl StdError for ResponseDeserializationError {}
 
 /// An error was reported by the server (HTTP status code 4xx or 5xx)
 #[derive(Debug)]
+#[non_exhaustive]
 pub enum ServerError<E> {
```

Comment:
I don't think this one will change (without a larger API-breaking refactoring), so it can be exhaustive.